### PR TITLE
Add Feature Workflow Issue Template

### PR DESCRIPTION
**Pull request by pwittrock**
## Contributor guidelines
1. Please read our [contributor guidelines](https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md).
2. See our [developer guide](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md).
3. Follow the instructions for [labeling and writing a release note for this PR](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes) in the block below. (_this would also contain updated detail later about this new template_)

``` release-note
* Use the release-note-* labels to set the release note state 
* Clear this block to use the PR title as the release note 
-OR-
* Enter your extended release note here
```

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @johndmulhausen @bgrant0607 

Does the template seem to heavy?  Should we cut it down to just api review + docs review?

File: docs/devel/feature-flow-issue-template.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Large features typically go through three stages: [Alpha, Beta, and Stable](https://github.com/kubernetes/kubernetes/blob/master/docs/api.md#api-versioning)
+Each stage requires various approvals from various teams.
+
+You can use the checkboxes below to
+track your progress.  You can delete the text above once you have read it.  Please keep the checklist.
+
+- Before Alpha
+    - [ ] Design Approval
+      - [ ] Propose your feature.  See [Advice for proposing features].  **PROPROSAL-ISSUE-NUMBER**
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: I think line 12 is a subset of line 13.  Delete line 12?

File: docs/devel/feature-flow-issue-template.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - [ ] Start with a < 1 page description of design in an issue comment below this. [advice for design description]  **1PAGER-ISSUE-NUMBER**
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: Feedback was that proposals should be done with PRs, not issues, for commenting purposes.  So, the action here is submit a PR with a proposal in docs/proposals/foo.md.

File: docs/devel/feature-flow-issue-template.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - [ ] Initial design review  		     [advice for getting a review from a team]  **DESIGN-REVIEW-ISSUE-NUMBER**
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: I just realized that we could use docs/proposals/OWNERS to ensure proposals are reviewed by appropriate people.  I don't think we need a separate line item for "Initial Design review".

File: docs/devel/feature-flow-issue-template.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      - [ ] Initial API review (if API) **API-REVIEW-ISSUE-NUMBER**
+        -  cc @kubernetes/api
+    - [ ] Write (code + tests + docs) then get them merged.  **LIST-OF-PRS**
+      - [ ] **Code needs to be disabled by default.**   Verified by code OWNERS
+      - [ ] OWNER approval for code
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: Implied by the fact that the code is merged.

Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: So remove this line.

File: docs/devel/feature-flow-issue-template.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  - [ ] detailed user docs and examples
+    - **cc @kubernetes/docs**
+
+*FEATURE_STATUS: INCOMPLETE*
+
+More advice:
+
+Design
+   - If the feature is not simple, the approver might ask you to write a longer design doc.
+   - If there is a design doc, that goes in a PR under github.com/kubernetes/kubernetes/docs/proposals.  Link that PR on this issue.
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: Lets make this required for all new features.

File: docs/devel/feature-flow-issue-template.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  - [ ] User docs with tutorials
+        - *Updated walkthrough / tutorial* in the docs repo: kubernetes/kubernetes.github.io
+        - cc @kubernetes/docs on docs PR
+        - **cc @kubernetes/feature-reviewers on this issue** to get approval before checking this off
+  - [ ] Thorough API review
+    - cc @kubernetes/api
+- Before Stable
+  - [ ] Soak, load testing 			
+  - [ ] detailed user docs and examples
+    - **cc @kubernetes/docs**
```
Comment (erictune) [inline docs/devel/feature-flow-issue-template.md]: docs/proposals/foo.md needs to move to docs/design/foo.md and be updated to reflect how we did it.  Design docs are required to be maintained separately and in addition to user docs.

**Comment by bgrant0607** [comment]
We should just create the new repo and create the template there.

**Comment by bgrant0607** [comment]
cc @quinton-hoole

**Comment by erictune** [comment]
@bgrant0607 the thought was to try out the template with a few features before moving to a new repo.

File: docs/devel/feature-flow-issue-template.md (hunk 1 above)
Comment (pwittrock) [inline docs/devel/feature-flow-issue-template.md]: Done

File: docs/devel/feature-flow-issue-template.md (hunk 2 above)
Comment (pwittrock) [inline docs/devel/feature-flow-issue-template.md]: Done

File: docs/devel/feature-flow-issue-template.md (hunk 3 above)
Comment (pwittrock) [inline docs/devel/feature-flow-issue-template.md]: Done

File: docs/devel/feature-flow-issue-template.md (hunk 5 above)
Comment (pwittrock) [inline docs/devel/feature-flow-issue-template.md]: Done

File: docs/devel/feature-flow-issue-template.md (hunk 6 above)
Comment (pwittrock) [inline docs/devel/feature-flow-issue-template.md]: Done

**Comment by pwittrock** [comment]
Moved to https://github.com/kubernetes/features/pull/1

**Pull request closed without merging**